### DISCUSSION TITLE
Multiple fields on the same row

### DISCUSSION THREAD

**Discussion by snvgglebear** [Budibase/budibase#6158] [Help] [answered]
Is it possible to put multiple form fields on the same row? If so how?

**Comment by melohagan** [comment]
Like this? https://docs.budibase.com/docs/designing-horizontal-forms

**Reply by snvgglebear** [reply]
Are sections implemented as tables under the hood? I am having a problem where I cannot get all items in a 3 column section to fit on the same row. Is it related to also have 2 column sections in the same field group?

**Reply by melohagan** [reply]
Hey @gondilon 

Sections are just containers for layout. Would you be able to provide some screenshots of your app and component tree?

Or even the app export and I could have a look for you.

**Reply by kpturner** [reply]
The link provided is broken

**Comment by melohagan** [comment]
Thanks @kpturner 

To answer the OP with a little more information, you can do this using a [Container](https://docs.budibase.com/docs/container)

1. Add your Form component
2. Add a container with *Row Layout*
3. Add a number of fields to your container (for example three)
4. Duplicate as needed

![Screenshot 2023-07-03 at 13 18 40](https://github.com/Budibase/budibase/assets/101575380/ae6bddb1-40fa-48f8-8e5d-546914e013f8)

**Reply by kpturner** [reply]
Could you define what you mean by `Row Layout`?
The `Direction` on a `Container` has two options (it would be useful all these layout icons had tooltips by the way).
The docs says
> _Change the direction of the components nested within your container. You can layout your components in a row (→) or a column (↓)._


The first option seems to stack fields one on top of the other, and the second places them side by side.

**Reply by melohagan** [reply]
Good point, I'll update those docs when I get a moment. 
![Screenshot 2023-07-03 at 13 28 40](https://github.com/Budibase/budibase/assets/101575380/1c4ffca0-f5e1-4a15-98d7-bef5ee5c69cc)

**Reply by kpturner** [reply]
So if I want those three fields to take up a third of the width each, what is the recommendation?  Adding three without any changes does this
![image](https://github.com/Budibase/budibase/assets/308646/bc4766e1-31ee-4e73-85e1-a7c021e71e75)

**Reply by melohagan** [reply]
You can set the width of each field to 100% under the styles tab. I'm looking now to see if I can incorporate the Section component into the field group to make this a little easier.

**Reply by kpturner** [reply]
Changing the widths to 100% seems to make no difference at all for me :(

**Reply by kpturner** [reply]
And if I try the same with options pickers (from a data provider) it looks distinctly worse.   These pickers have 50% width
![image](https://github.com/Budibase/budibase/assets/308646/6b748745-a6e5-40cb-b9c0-12c33857c375)

**Reply by melohagan** [reply]
Yes the formatting does seem to be a little strange. I had more luck setting the form field width to 200% instead of 100%.

**Reply by kpturner** [reply]
Putting each field in a `Field group` yields marginally better results, but not exactly right.  It seems intent on restrictying the widths of the pickers
![image](https://github.com/Budibase/budibase/assets/308646/275de849-c031-4373-a7eb-0c5d4882395c)

**Reply by kpturner** [reply]
I have tried duplicating three containers and weirdly 2 look OK and one does not. 

Not sure why yet though

![image](https://github.com/Budibase/budibase/assets/308646/2e715399-c167-4beb-b029-2e3a910558d2)

**Comment by kpturner** [comment]
Finally got it to work like this
![image](https://github.com/Budibase/budibase/assets/308646/ef57d67c-64e9-46eb-9995-7238a5122c16)

![image](https://github.com/Budibase/budibase/assets/308646/2577b321-6804-486a-aa5c-e601a08138a6)

**Reply by melohagan** [reply]
Nice one! I also have a PR out for making this a bit easier: https://github.com/Budibase/budibase/pull/11110

**Reply by jasonford1** [reply]
Thank you for writing this up. If I may kindly suggest marking this as the answer. I was experiencing the exact same issue and didn't solve until this exact reply.

**Comment by melohagan** [comment, marked as answer]
Hey @jasonford1 

That answer is a little out-dated. The best way to do this going forward is shown here: 

- https://docs.budibase.com/docs/display-mutiple-fields-in-one-row
- https://docs.budibase.com/docs/field-groups#layout
